### PR TITLE
PLANNER-2870 Environments should be configurable via branch config

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/PLANNER-2870

Related:
- https://github.com/kiegroup/kogito-pipelines/pull/801
- https://github.com/kiegroup/optaplanner/pull/2550
- https://github.com/kiegroup/optaweb-vehicle-routing/pull/842
- https://github.com/kiegroup/optaplanner-quickstarts/pull/505
- https://github.com/kiegroup/optaplanner-website/pull/569
- [9.x] https://github.com/kiegroup/optaplanner/pull/2558

### CODE REVIEW

File: .ci/jenkins/config/branch.yaml [hunk 1]
```diff
@@ -1,78 +1,110 @@
...
+      QUARKUS_NATIVE_BUILDER_IMAGE: mandrel
+      ADDITIONAL_TIMEOUT: 720
+    ids:
+    - native
+    - prod
+    - lts
+  kogito-bdd:
+    auto_generation: false
+    ids:
+    - cloud
```
Comment (triceo) [inline .ci/jenkins/config/branch.yaml:51]: Guessing this is not necessary for OptaPlanner?

File: .ci/jenkins/config/branch.yaml [hunk 2]
```diff
@@ -1,78 +1,110 @@
...
   image:
-    registry_credentials: optaplanner-quay-token
+    registry_credentials: tradisso_registry_token
     registry: quay.io
-    namespace: optaplanner
+    namespace: tradisso
     latest_git_branch: main
 jenkins:
-  email_creds_id: OPTAPLANNER_CI_EMAIL
+  email_creds_id: KOGITO_CI_EMAIL_TO_PERSO
```
Comment (triceo) [inline .ci/jenkins/config/branch.yaml:107]: Kogito?

File: .ci/jenkins/config/main.yaml [hunk 3]
```diff
@@ -6,23 +6,16 @@ ecosystem:
...
       author:
-        name: kiegroup
+        name: radtriste
         credentials_id: kie-ci
-      branch: main
+      branch: planner-2870-refactor-envs
     path: .ci/jenkins/config/branch.yaml
 jenkins:
-  email_creds_id: OPTAPLANNER_CI_EMAIL
+  email_creds_id: KOGITO_CI_EMAIL_TO_PERSO
```
Comment (triceo) [inline .ci/jenkins/config/main.yaml:21]: Kogito?

File: .ci/jenkins/config/branch.yaml (hunk 2 above)
Comment (radtriste) [reply .ci/jenkins/config/branch.yaml:107]: test commit. That is why I am still on draft status :)

File: .ci/jenkins/config/branch.yaml (hunk 1 above)
Comment (radtriste) [reply .ci/jenkins/config/branch.yaml:51]: indeed will remove that one

**Review by rsynek — APPROVED** [review]

**Review by triceo — APPROVED** [review]

**Pull request merged**